lib/features/run_placed/presentation/ui/screen/run_placed_screen.dart

```diff
@@ -1,58 +1,77 @@
 import 'dart:convert';
 
+import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:hotfoot/features/navigation_screen/presentation/bloc/navigation_screen_bloc.dart';
 import 'package:hotfoot/features/run_placed/presentation/ui/widgets/accept_delivery_button.dart';
 import 'package:hotfoot/features/run_placed/presentation/ui/widgets/cancel_delivery_button.dart';
 import 'package:hotfoot/features/run_placed/presentation/ui/widgets/open_close_chat_button.dart';
+import 'package:hotfoot/features/runs/domain/use_cases/get_run_stream.dart';
+import 'package:hotfoot/injection_container.dart';
 
 class RunPlacedScreen extends StatelessWidget {
+
+  Stream<QuerySnapshot> returnCurrentRunStream (GetRunStream useCase, String id) async* {
+    final result = await useCase.call(id);
+    if (result.isRight()) {
+      yield* result.getOrElse(null);
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
     final navScreenBloc = BlocProvider.of<NavigationScreenBloc>(context);
     final currRun = navScreenBloc.state.runModel;
+    final GetRunStream useCase = GetRunStream(runsRepository: sl(),);

```

Comment:
Rename `useCase` to `getRunStream` ? I think we had only used `useCase` in the unit tests.